src/dynamic_array.c

```diff
@@ -47,3 +45,37 @@ void iter_seq(DynamicArray *dy_arr) {
         printf("%d", dy_arr->data[i]);
     };
 };
+
+void insert_at(DynamicArray *dy_arr, size_t pos, int value) {
+    if (dy_arr->size == dy_arr->capacity) {
```

Comment:
when you insert the values, for this function only 1 value, needed size for this operation is 
new_size = dy_arr->size + count_of_new_element(1). Than check is capacity enough, otherwise resize. And this check shouldn`t be
if (dy_arr->size == dy_arr->capacity) {
it should be 
if (new_size > dy_arr->capacity) { than resize(), because not enough memory.